spark3/src/main/java/org/apache/iceberg/spark/source/SparkFilesScan.java

```diff
@@ -62,6 +63,9 @@
     long tableSplitSize = PropertyUtil.propertyAsLong(props, SPLIT_SIZE, SPLIT_SIZE_DEFAULT);
     this.splitSize = Spark3Util.propertyAsLong(options, SparkReadOptions.SPLIT_SIZE, tableSplitSize);
 
+    Preconditions.checkArgument(splitSize > 0,
```

Comment:
I think we better move this validation to TableScanUtil and also validate other args such as `lookback`.